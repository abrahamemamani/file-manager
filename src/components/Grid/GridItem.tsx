import React, { FC, ReactNode } from "react";

export interface GridItemProps {
  children: ReactNode;
  className?: string | undefined;
}

export const GridItem: FC<GridItemProps> = ({ children, className = "" }) => (
  <div className={`${className}`}>{children}</div>
);
