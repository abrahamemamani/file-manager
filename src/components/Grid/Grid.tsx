import React, { FC, ReactNode } from "react";
import { GridItem } from "./GridItem";

export interface GridProps {
  children: ReactNode;
  className?: string | undefined;
}

const GridComponent: FC<GridProps> = ({ children, className = "" }) => (
  <div className={`grid ${className}`}>{children}</div>
);

GridComponent.displayName = "Grid";
GridItem.displayName = "Grid.Item";

export const Grid = Object.assign(GridComponent, {
  Item: GridItem,
});
