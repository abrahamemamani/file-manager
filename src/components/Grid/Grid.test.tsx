import { describe, expect, it } from "vitest";
import { Grid, GridProps } from "./Grid";
import { render } from "@testing-library/react";

const defaultProps: GridProps = {
  children: <div></div>,
  className: "class",
};

describe("[components/Grid]", () => {
  describe("when Grid is rendered", () => {
    it("should render", () => {
      const { asFragment } = render(<Grid {...defaultProps} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
