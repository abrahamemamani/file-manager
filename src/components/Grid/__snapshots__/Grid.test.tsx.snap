// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[components/Grid] > when Grid is rendered > should render 1`] = `
<DocumentFragment>
  <div
    class="grid class"
  >
    <div />
  </div>
</DocumentFragment>
`;
